### ISSUE TITLE
WMR - Every game just pause itself after latest SteamVR update - Controllers issue ?

### ISSUE THREAD

**Issue by kakek** [LibreVR/Revive#1380]
**Describe the bug**
Aasgard wrath is stuck at game paused. Journey of the gods is stuck at a similar message. I suspect controller issue.

**To Reproduce**

**Environment (please complete the following information):**
 - App: Aasgard wrath and journey of the gods.
 - OS: win10
 - Headset: WMR

**Versions (please complete the following information):**
 - Revive: 1.9.1.5
 - SteamVR: 1.8.1.9 OR 1.8.2, latest from main branch and beta.

**Additional context**
Problem already noted in issue 1356 ( https://github.com/LibreVR/Revive/issues/1356 )
Solution was to switch out of steamVR beta, so the issue was closed.
Problem is now on the main branch as well, so need either a new issue, or reopening hte old one.

I noticed that there is also a problem with rebinding controllers via steamVR in steam games. Although they work, I can't rebind my controllers because steamVR tells me there's no controller detected.
I suspect the 2 issue might be linked since they appeared with the same steamVR update.

**Comment by madpat3** [comment]
i have this Problem, too. is there no Chance to roll back to steam vr 1.7?

**Comment by kakek** [comment]
OK, it seems to be fixed for me by uninstalling steamVR, mixed reality, every steamVR redistribuable, and revive.
Then deleting manually any leftover files ( I had a few, specially since I installed various version of both steamVR and revive on different paths at some point, so I made sure to delete every file on any disk I had it installed at some point )
Then reinstalled everything one by one.

And now it's good.

Could you try that and see if it works for you ?

**Comment by madpat3** [comment]
first i deinstalled, then reinstalled "revive" only… didn't work. now deinstalled and reinstall steam-vr. takes a while (didn't know it was 1.6 GB big ;) standby...

**Comment by madpat3** [comment]
didn't work, either. now i'm doing it your way. deinstalled every data from mixed reallity over steam vr to revive. Right now i'm reinstalling mixed reallity. next will be steam vr and then revive. standby...

**Comment by madpat3** [comment]
so i tried the same method like you, but it didn't work for me.

**Comment by kakek** [comment]
And now it doesn't work for me either anymore.
Somehow, the problem came back. As far as I can tell I have changed NOTHING.

So yeah, the issue is still opened.

**Comment by madpat3** [comment]
so i've got several oculus games:

vader immortal 1&2
blade runner 2049
abe vr
technolust
asgard's wrath
oculus home
oculus touch basics
oculus dreamdeck
oculus contact

all are running with revive pretty well, exept for asgard's wrath (with the mentioned Problem).

**Comment by Puni12345** [comment]
Also have this problem sometimes - for me the solution is to press the windows menu button on the controller short once. Then the overlay of wmr gets displayed. Then i press the wmr menu button again - once short - then the game is unpaused.

**Comment by kakek** [comment]
You're right, it only concern asgaard wrath and journey of the gods. Those were the 2 games I had installed at the time, and since they where totally different from different timeframe, I figured it must affect other games.
Turns out I was just unlucky enough that the 2 games I tried were the only 2 with this problem.

And switching the WMr interface changed nothing for me.

However I might have something else that helps.
this might just be linked to the fact that Revive changed the location of the binding.
Try putting the files in that zip into C:\Users\*yourname*\AppData\Roaming\Revive, and in C:\Users\*yourname*\AppData\Local\Revive

[Input.zip](https://github.com/LibreVR/Revive/files/3865337/Input.zip)

For me it seems to work ? Require more testing.

**Comment by madpat3** [comment]
I'll try that zip when time allows. 
meanwile you should maybe change the title of this issue... not to confuse people.

**Comment by sergeylamzin** [comment]
@kakek Yes! Thank you!

This worked for me on Asgard's Wrath.
SteamVR - 1.8.20
Win10
Valve Index

Just to be on the safe side I copied 
C:\Program Files\Revive\Input
into
C:\Users*yourname*\AppData\Roaming\Revive
and
C:\Users*yourname*\AppData\Local\Revive

**Comment by madpat3** [comment]
i don't have no Revive-folder inside the roaming-folder !?

**Comment by madpat3** [comment]
because i don't have no C:\Users*yourname*\AppData\Roaming\Revive Folder, i just put in the input-Folder into the C:\Users*yourname*\AppData\Local\Revive path. but that didn't work.

**Comment by kakek** [comment]
Might sound stupid, but try creating the other directory manualy and copying the files. I don't think it'll work, but hey, it's 15 seconds ... might as well give it a shot.

**Comment by madpat3** [comment]
didn't work. tried it with the files alone in the specific folders and all in one folder named "Input". no success.

**Comment by Ma-ciejCh** [comment]
I have the same issue in Asgard's Wrath. Not 100% sure but it seems to be connected to tracking of my headset, game sometimes tends to pause when I look at the ground and unpause when I look towards a tracking station - I may be wrong on that. The pause screen just says "game paused". Other Revive games work fine.

The suggestion to put files in the AppData/... directories didn't help.

Software:
Revive 1.9.1, SteamVR 1.8.21 (also present with beta 1.9.5), Windows 10 Pro **N** 1909 (the educational version)

Hardware:
Valve Index, knuckle controllers, lighthouse stations 2.0
i5-4440, RTX 2060

**Comment by Ma-ciejCh** [comment]
With Revive 1.9.2 and SteamVR 1.9.15 the issue is no longer present for me.

**Comment by madpat3** [comment]
that Maybe, but this issue concerns wmr-headsets. Valve index is a different matter, though.

**Comment by madpat3** [comment]
will this ever be fixed?
i mean, at least you could say: "will not be fixed, we're too buisy.", or something.

**Comment by CrossVR** [comment]
will not be fixed, we're too buisy answering issues where people are demanding comments from the developers.

**Comment by CrossVR** [comment]
This is a bug in the WMR driver, not in Revive, but perhaps this workaround is enough: https://ci.appveyor.com/project/LibreVR/revive/builds/30614493/artifacts

**Comment by madpat3** [comment]
Give this man the Crown, that he deserves, for he is the true king and no one else ;) !!!

Yes this works. So after a pause of over 4 Month all WMR users can finally play those premium titles again.

Thank you Jules,
and hopefully this CrossVR thing, will become a thing in the future, so we will not have to suffer such issues with incompatibillity between different platforms, any longer.

Thanks to you and your kinds.

Best regards
Patrick

**Comment by CrossVR** [comment]
No need for all that praise, just ask nicely next time. Something like: "I'm still having this issue, could someone please look into it?" will get a much less sarcastic response.

**Comment by madpat3** [comment]
understood. no harm meant, anyway.
thank you for your unpayed work, doing many people a favour.

**Comment by aman-agrawala** [comment]
This also fixes vader immortal for me. 

Software:
SteamVR 1.10.12, Windows 10 1903

Hardware:
WMR
i5-4690K, R9 380

**Comment by we7** [comment]
> This is a bug in the WMR driver, not in Revive, but perhaps this workaround is enough: https://ci.appveyor.com/project/LibreVR/revive/builds/30614493/artifacts

I'm sorry for replying to an old comment but i didn't know how to use this solution can you explain it ?
